src/metadatas.rs

```diff
@@ -10,28 +10,55 @@ use std::collections::HashMap;
 pub struct Metadata {
     pub start_date: Option<String>,
     pub end_date: Option<String>,
-    pub stops_count: usize,
-    pub stop_areas_count: usize,
-    pub stop_points_count: usize,
-    pub stops_with_wheelchair_info_count: Option<usize>,
-    pub lines_count: usize,
-    pub trips_count: usize,
-    pub trips_with_bike_info_count: usize,
-    pub trips_with_wheelchair_info_count: usize,
-    pub trips_with_shape_count: usize,
     pub networks: Vec<String>,
     pub networks_start_end_dates: Option<HashMap<String, Option<Interval>>>,
     pub modes: Vec<String>,
     pub issues_count: std::collections::BTreeMap<IssueType, usize>,
     pub has_fares: bool,
     pub has_shapes: bool,
     pub has_pathways: bool,
-    pub lines_with_custom_color_count: usize,
     // some stops have a pickup_type or drop_off_type equal to "ArrangeByPhone"
     pub some_stops_need_phone_agency: bool,
     // some stops have a pickup_type or drop_off_type equal to "CoordinateWithDriver"
     pub some_stops_need_phone_driver: bool,
     pub validator_version: String,
+    pub stops_count: usize,
+
+    pub stats: Stats,
+    // legacy fields, now counter should be in stats
+    pub stop_areas_count: usize,
+    pub stop_points_count: usize,
+    pub stops_with_wheelchair_info_count: Option<usize>,
+    pub lines_count: usize,
+    pub trips_count: usize,
+    pub trips_with_bike_info_count: usize,
+    pub trips_with_wheelchair_info_count: usize,
+    pub trips_with_shape_count: usize,
+    pub lines_with_custom_color_count: usize,
+    pub trips_with_trip_headsign_count: usize,
+}
+
+#[derive(Serialize, Debug)]
+pub struct Stats {
+    pub stops_count: usize,
+    pub stop_areas_count: usize,
+    pub stop_points_count: usize,
+    pub stops_with_wheelchair_info_count: Option<usize>,
+
+    pub lines_count: usize,
+    pub lines_with_custom_color_count: usize,
+    pub lines_with_short_name_count: usize,
+    pub lines_with_long_name_count: usize,
```

Comment:
Use the `routes_` prefix?

Reply:
some old transmodel habits :sweat_smile: 

I'll rename them